core/src/main/scala/com/microsoft/azure/synapse/ml/causal/LinearDMLEstimator.scala

```diff
@@ -0,0 +1,279 @@
+    )
+    val (outcomeEstimator, outcomeResidualPredictionColName, outcomePredictionColsToDrop) = getModel(
+      getOutcomeModel.copy(getOutcomeModel.extractParamMap()), getOutcomeCol, Array(getTreatmentCol)
+    )
+
+    val treatmentResidualVecCol = DatasetExtensions.findUnusedColumnName("treatmentResidualVec", dataset)
+
+    def setUpDMLPipeline(train: Dataset[_], test: Dataset[_]): DataFrame = {
+      val treatmentModel = treatmentEstimator.fit(train)
+      val outcomeModel = outcomeEstimator.fit(train)
+
+      val treatmentResidual =
+        new ResidualTransformer()
+          .setObservedCol(getTreatmentCol)
+          .setPredictedCol(treatmentResidualPredictionColName)
+          .setOutcomeCol(SchemaConstants.TreatmentResidualColumn)
+      val dropTreatmentPredictedColumns = new DropColumns().setCols(treatmentPredictionColsToDrop.toArray)
+      val outcomeResidual =
+        new ResidualTransformer()
+          .setObservedCol(getOutcomeCol)
+          .setPredictedCol(outcomeResidualPredictionColName)
+          .setOutcomeCol(SchemaConstants.OutcomeResidualColumn)
+      val dropOutcomePredictedColumns = new DropColumns().setCols(outcomePredictionColsToDrop.toArray)
+      val treatmentResidualVA =
+        new VectorAssembler()
+          .setInputCols(Array(SchemaConstants.TreatmentResidualColumn))
+          .setOutputCol(treatmentResidualVecCol)
+          .setHandleInvalid("skip")
+      val pipeline = new Pipeline().setStages(Array(
+        treatmentModel, treatmentResidual, dropTreatmentPredictedColumns,
+        outcomeModel, outcomeResidual, dropOutcomePredictedColumns,
+        treatmentResidualVA))
+
+      pipeline.fit(test).transform(test)
+    }
+
+    // Note, we perform these steps to get ATE
+    /*
+      1. Split sample, e.g. 50/50
+      2. Use the first split to fit the treatment model and the outcome model.
+      3. Use the two models to fit a residual model on the second split.
+      4. Cross-fit by fitting the treatment and outcome models with the second split and the residual model with the first split.
+      5. Average slopes from the two residual models.
+    */
+    val splits = dataset.randomSplit(getSampleSplitRatio)
+    val (train, test) = (splits(0).cache, splits(1).cache)
+    val treatmentEffectDFV1 = setUpDMLPipeline(train, test)
+    val treatmentEffectDFV2 = setUpDMLPipeline(test, train)
+
+    // Average slopes from the two residual models.
+    val regressor = new GeneralizedLinearRegression()
+      .setLabelCol(SchemaConstants.OutcomeResidualColumn)
+      .setFeaturesCol(treatmentResidualVecCol)
+      .setFamily("gaussian")
+      .setLink("identity")
+      .setFitIntercept(false)
+    val (lrmV1, lrmV2) = (regressor.fit(treatmentEffectDFV1), regressor.fit(treatmentEffectDFV2))
+    Seq(train, test).foreach(_.unpersist)
+
+    val ate = Seq(lrmV1, lrmV2).map(_.coefficients(0)).sum / 2
```

Comment:
sorry I didn't get this, "remove the "Linear" prefix"

Reply:
> This looks like it could be generalized to any SparkML model allowing you to remove the "Linear" prefix in this class. Though dont worry if you think about this and deem it impossible because of other issues

The naming follows the naming in EconML package. I believe the word "Linear" refers to the linear regressor (GeneralizedLinearRegression) in the last stage. We will double check with EconML team about that.

Reply:
we decided to use name "DoubleMLEstimator", thanks for the feedback!